hal/ia32/acpi.c

```diff
@@ -79,12 +79,23 @@ static void _hal_acpiFadtHandler(hal_syspage_t *hs, sdt_header_t *header)
 	}
 }
 
+
+static void _hal_acpiHpetHandler(hal_syspage_t *hs, sdt_header_t *header)
+{
+	if (_hal_checksumVerify(header, header->length) != 0) {
+		hs->hpet = (unsigned int)header;
```

Comment:
`hs->hpet` should be of `addr_t` type

Reply:
Ok, at least go with `unsigned long` + `FIXME` for now

Reply:
Why `unsigned long`? `addr_t` is defined as `u32` which is defined as `unsigned int`

Reply:
In case we try to make a x64 port in the meantime

Reply:
Perphaps `FIXME - should be addr_t`?